### PR TITLE
EAS: add STS client logging

### PR DESCRIPTION
**Pull request by bpdohall**
This PR adds additional logging to the AWS STS client used by the External Audit Storage integration.

Previously, all errors for AWS API were swallowed and failures to obtain credentials were difficult to troubleshoot.

With this change an unsuccessful call to `sts.AssumeRoleWithWebIdentity` will emit a log entry containing a message similar to the following:

```
Response
HTTP/1.1 400 Bad Request
Connection: close
Content-Length: 276
Content-Type: text/xml
Date: Wed, 09 Oct 2024 17:19:04 GMT
X-Amzn-Requestid: GUID

<ErrorResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
  <Error>
    <Type>Sender</Type>
    <Code>InvalidIdentityToken</Code>
    <Message>Incorrect token audience</Message>
  </Error>
  <RequestId>GUID</RequestId>
</ErrorResponse>
```

The log entry will also include fields identifying the role being requested, the name of the integration and an indicator whether the cluster is configured to use an S3 bucket as the source for OIDC `/.well-known/*` materials.

### CODE REVIEW

File: lib/integrations/externalauditstorage/configurator.go [hunk 1]
```diff
@@ -118,6 +127,27 @@ func (o *Options) setDefaults(ctx context.Context, region string) error {
 	return nil
 }
 
+func awsLoggerFunc(fields logrus.Fields) func(classification logging.Classification, format string, v ...interface{}) {
```
Comment (nklaassen) [inline lib/integrations/externalauditstorage/configurator.go:130]: you might have to also cast the func before returning it, but i think this makes the function signature more clear and eliminates the type case where it's called
```suggestion
func awsLoggerFunc(fields logrus.Fields) logging.LoggerFunc {
```

File: lib/integrations/externalauditstorage/configurator.go [hunk 2]
```diff
@@ -109,6 +111,13 @@ func (o *Options) setDefaults(ctx context.Context, region string) error {
 			config.WithRegion(region),
 			config.WithUseFIPSEndpoint(useFips),
 			config.WithRetryMaxAttempts(10),
```
Comment (nklaassen) [inline lib/integrations/externalauditstorage/configurator.go:113]: hmm... before we try all this request/response body logging... do you mind just deleting this `WithRetryMaxAttempts` line? I had copied it from some WIP code tobiasz had, and I honestly don't know the reason for it. I bet it's actually what's causing us to go over the 30s context deadline and just get useless errors. Hopefully we get good error messages without it 🤞 I'm pretty sure there's no good reason at all for us to have it

looks like the default max attempts is 3 with a max backoff of 4 seconds (i may be off by a factor of 2 but it should be under 30s for sure)

Comment (bpdohall) [reply lib/integrations/externalauditstorage/configurator.go:113]: You're correct. When I remove the max attempts lines, log message switches over to: 
```
Failed to retrieve new credentials: failed to retrieve credentials, operation error STS: AssumeRoleWithWebIdentity, exceeded maximum number of attempts, 3, https response error StatusCode: 400, RequestID: GUID, InvalidIdentityToken: Incorrect token audience
```

**Comment by bpdohall** [comment]
Abandoning this approach in favour of removing `WithRetryMaxAttempts`

**Pull request closed without merging**